shell/common/animator.cc

```diff
@@ -99,7 +99,10 @@ void Animator::BeginFrame(
   FML_DCHECK(producer_continuation_);
   const fml::TimePoint frame_target_time =
       frame_timings_recorder_->GetVsyncTargetTime();
-  dart_frame_deadline_ = frame_target_time.ToEpochDelta();
+  dart_frame_deadline_ =
+      frame_target_time.ToEpochDelta() -
+      ((frame_target_time - frame_timings_recorder_->GetVsyncStartTime()) /
+       layer_tree_pipeline_->inFlightCount());
```

Comment:
Please comment the intention of this calcuation

Reply:
This way is still incorrect in most cases. I may need to add another timestamp returned from vsync callback or maybe some new method on the `VsyncWaiter` itself to differentiate when `frame_target_time` will be after the next vsync firing time, so the full time diff is not available for use.

Reply:
I have updated this to a new approach, the Vsync callback will now contain both `vsync_target_time` (wall clock budget) and `next_vsync_start_time` (single thread budget)